TF/src/common/Graph.py

```diff
@@ -0,0 +1,137 @@
+        # List to store visit status of a node
+        visited = [False] * (len(self.nodes))
+        queue = Queue()
+
+        # For each start node, if not already visited, start a new traversal
+        for node_index in self.start_node_indices:
+            queue.put(node_index)
+            visited[node_index] = True
+            
+            # BFS
+            while not queue.empty():
+                src_node_index = queue.get()
+
+                if src_node_index not in self.adj_list:
+                    continue
+
+                for [edge_index, dest_node_index] in self.adj_list[src_node_index]:
+                    print(
+                        self.nodes[src_node_index].label, 
+                        self.edges[edge_index].label, 
+                        self.nodes[dest_node_index].label,
+                    )
+
+                    if not visited[dest_node_index]:
+                        visited[dest_node_index] = True
+                        queue.put(dest_node_index)
+
+    # Print nodes with attributes
+    def print_nodes(self):
+        print("\nOperators\n")
+        for node in self.nodes:
+            attrs = vars(node)
+            
+            for item in attrs.items():
+                if item[1] != None and item[1] != "NONE":
+                    print(str(item[0]) + ": " + str(item[1]), end=' ')
+            
+            print()
+
+    # Print edges with attributes
+    def print_edges(self):
+        print("\nTensors\n")
+        for edge in self.edges:
+            attrs = vars(edge)
+            
+            for item in attrs.items():
+                if item[1] != None and item[1] != "NONE":
+                    print(str(item[0]) + ": " + str(item[1]), end=' ')
+            
+            print()
+
+    # Function to get those edges which are traversed during BFS
+    # Assumption is that the tensors not reached by traversal are weights, bias etc.
+    def get_traversed_edges(self):
+        visited = dict()
+        queue = Queue()
+
+        # set to maintain unique edges encountered
+        traversed_edges = set()
+        
+        # List to store visit status of a node
+        for node_index in range(len(self.nodes)):
+            visited[node_index] = False
+
+        # For each start node, if not already visited, start a new traversal
+        for node_index in self.start_node_indices:
+            queue.put(node_index)
+            
+            # BFS
+            while not queue.empty():
+                source_node_index = queue.get()
+
+                if source_node_index not in self.adj_list:
+                    continue
+
+                for [edge_index, dest_node_index] in self.adj_list[source_node_index]:
+                    traversed_edges.add(self.edges[edge_index])
+
+                    if not visited[dest_node_index]:
+                        visited[dest_node_index] = True
+                        queue.put(dest_node_index)
+        
+        return list(traversed_edges)
```

Comment:
Same for other functions as well. See if we can reduce some code duplication. 'print_graph' and 'get_traversed_graph' is same and repeated for each TFLite and TF.

Reply:
Will try to address 'print_graph' and 'get_traversed_graph' code duplication in the subsequent commits.